packages/tpetra/core/ext/TpetraExt_MatrixMatrix_def.hpp

```diff
@@ -121,7 +122,9 @@ void Multiply(
 #ifdef HAVE_TPETRA_MMM_TIMINGS
   std::string prefix_mmm = std::string("TpetraExt ") + label + std::string(": ");
   using Teuchos::TimeMonitor;
-  TimeMonitor MM(*TimeMonitor::getNewTimer(prefix_mmm + std::string("MMM All Setup")));
+  //MM is used to time setup, and then multiply.
+  
+  RCP<TimeMonitor> MM = rcp(new TimeMonitor(*TimeMonitor::getNewTimer(prefix_mmm + std::string("MMM All Setup"))));
```

Comment:
Using `RCP` with a scope guard like TimeMonitor always makes me sad, especially since you use it correctly (without a scope guard) elsewhere in this commit.  (If the scope guard isn't appropriate for the situation, you could always call start and stop on the timer manually.)

Reply:
@mhoemmen If I want to use start() and stop(), do I use the Teuchos::Time directly, instead of TimeMonitor?

Reply:
@brian-kelley wrote:
> If I want to use start() and stop(), do I use the Teuchos::Time directly, instead of TimeMonitor?

Oh wait -- I realize now that TimeMonitor has the stacked timer option, so you should use the TimeMonitor object.  Sorry :(